Challenge 1 core concepts and styling

alfonsoramirzz opened the pull request:
Mini-Challenge 1: Core Concepts + Styling => complete

alfonsoramirzz committed:
File renamed to use jsx extension

alfonsoramirzz committed:
- Navbar was added.
- Vertical and horizontal menu were added.
- Videos section was added.

alfonsoramirzz committed:
Videos were added

erickwize commented on src/components/Video/Video.component.jsx:
```diff
@@ -0,0 +1,68 @@
+          <section className="preview-container">
+            <img src={image} alt="" />
+            <span className="time-status">5:15</span>
+            <div className="overlay-preview">
+              <div className="play-icon">
+                <svg
+                  viewBox="0 0 24 24"
+                  preserveAspectRatio="xMidYMid meet"
+                  focusable="false"
+                  className="style-scope yt-icon"
+                >
+                  <g className="style-scope yt-icon">
+                    <path d="M8 5v14l11-7z" className="style-scope yt-icon" />
+                  </g>
+                </svg>
+              </div>
+            </div>
+            <section className="content-container">
+              <h3 title={title}>{title}</h3>
+              <footer>
+                <p
+                  href="https://www.youtube.com/channel/UCukIPSb0N6_vRVD_5VlzqQg"
+                  target="_blank"
+                  className="channel-name"
+                  title={channelName}
+                >
+                  {channelName}
+                </p>
+              </footer>
+              <button className="more-button" type="button">
+                <svg
+                  viewBox="0 0 24 24"
+                  preserveAspectRatio="xMidYMid meet"
+                  focusable="false"
+                  className="style-scope yt-icon"
+                >
+                  <g className="style-scope yt-icon">
+                    <path
+                      d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
+                      className="style-scope yt-icon"
```
We encourage you to use Styled components instead of using CSS modules

alfonsoramirzz replied:
Hi Erick, I changed to use styled components 😃 thanks for your comment.

erickwize commented on src/components/Video/Video.styles.css:
```diff
@@ -0,0 +1,228 @@
+a {
```
Instead of having a big CSS file, we encourage you to have CSS in JS with styled-components and you could also use [Less syntax](http://lesscss.org/#nesting) to write less css.

alfonsoramirzz replied:
Hi Erick, I changed to use styled components 😃 thanks for your comment.

erickwize commented on src/components/core/AppContent/AppContent.component.jsx:
```diff
@@ -0,0 +1,9 @@
+import './AppContent.styles.css';
+
+function AppContent({ children }) {
+  return <div className="app-content scroll ">{children}</div>;
```
Same CSS module comment here

alfonsoramirzz replied:
Hi Erick, I changed to use styled components 😃 thanks for your comment.

erickwize commented on src/components/core/AppContent/AppContent.styles.css:
```diff
@@ -0,0 +1,30 @@
+.app-content {
```
Same CSS comment here

alfonsoramirzz replied:
Hi Erick, I changed to use styled components 😃 thanks for your comment.

erickwize commented on src/components/core/BlankPage/BlankPage.component.jsx:
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+function BlankPage() {
+  return <div className="blank" />;
```
Same CSS module comment here

erickwize commented on src/components/core/Menus/Horizontal.component.jsx:
```diff
@@ -0,0 +1,22 @@
+      </li>
+      <li>
+        <Link to="/" className="nav-links disabledCursor">
+          <span className="text-truncate ">Favorites</span>
```
Same CSS module comment here

erickwize commented on src/components/core/Menus/Vertical.component.jsx:
```diff
@@ -0,0 +1,27 @@
+      <li className="">
+        <Link to="/" className="nav-links disabledCursor">
+          <FontAwesomeIcon icon={faStar} />
+          <span className="text-truncate ">Favorities</span>
```
Same CSS module comment here

erickwize commented on src/components/core/Navbar/Navbar.component.jsx:
```diff
@@ -0,0 +1,47 @@
+            </span>
+          </li>
+        </ul>
+      </nav>
```
Same CSS module comment here

erickwize commented on src/components/core/Navbar/Navbar.component.jsx:
```diff
@@ -0,0 +1,47 @@
+
+      <nav className="navbar-responsive-container" ref={mainNavRef}>
+        <MenuHorizontal />
+      </nav>
```
If you used [styled-components you could use the props](https://styled-components.com/docs/basics#adapting-based-on-props) in the component to render things differently.

erickwize commented on src/components/core/Navbar/Navbar.styles.css:
```diff
@@ -0,0 +1,101 @@
+.navbar-container {
```
Same CSS modules comment here

erickwize commented on src/components/core/Search/Search.component.jsx:
```diff
@@ -0,0 +1,17 @@
+    <div className="search">
+      <input className="search-box" type="text" name="search" placeholder="Search..." />
+      <FontAwesomeIcon icon={faSearch} className="search-icon" />
+    </div>
```
Same CSS modules comment here

erickwize commented on src/components/core/Search/Search.styles.css:
```diff
@@ -0,0 +1,45 @@
+.search {
```
Same CSS modules comment here

erickwize commented on src/components/core/Sidebar/Sidebar.component.jsx:
```diff
@@ -0,0 +1,41 @@
+          </Link>
+        </li>
+      </ul>
+      <div className="shadow-bottom" />
```
Same CSS modules comment here

erickwize commented on src/components/core/Sidebar/Sidebar.styles.css:
```diff
@@ -0,0 +1,117 @@
+.navbar-aside-container {
```
Same CSS modules comment here

erickwize commented on src/pages/Home/Home.page.jsx:
```diff
@@ -1,37 +1,44 @@
+    <section className="homepage">
+      <div className="row">
+        {videoList.slice(1).map((v) => (
+          <div className="col" key={v.id.videoId.toString()}>
```
Same CSS modules comment here

erickwize commented on src/pages/Home/Home.styles.css:
```diff
@@ -1,8 +1,19 @@
-.homepage {
-  text-align: center;
+.col {
```
Same CSS modules comment here

alfonsoramirzz replied:
Hi Erick, I changed to use styled components 😃 thanks for your comment.

erickwize reviewed (changes requested):
## Comments (Mini-challenge 1 & 2)
Good work with the UI and the CSS the app looks very good! Also, nice work adding the hover animation to the sidebar and the elements in the sidebar!
I see you still don't have tests, we encourage you to start doing some tests for your components. You can use [React testing library](https://testing-library.com/docs/react-testing-library/intro/).
I recommend watching the styling part of the course, the styled-components were not implemented in the deliverable.

## Acceptance Criteria (Mini-challenge 1)
- [x] The header is rendered correctly.
- [x] A list of videos obtained from a mock file is displayed in the **Home View**.
- [x] CSS styles are applied correctly using the styled-components approach.

## Bonus Points (Mini-challenge 1)
- [x] No warnings or errors are logged in the browser console.
- [x] The UI is responsive

## Acceptance Criteria (Mini-challenge 2)
- [ ] Meaningful test cases were implemented for components and sub-routines logic.
- [ ] All the test cases were successful.

## Bonus Points (Mini-challenge 2)
- [ ] Test coverage is above 60%.

alfonsoramirzz committed:
- Global style was added.
- Sidebar, Brand,  were changed to styled component.
- eslint for testing library and jest dom plugins were added.
- ThemeProvider was added.
- Testing files for Sidebar, Brand and VerticalMenu were added.

alfonsoramirzz committed:
Test were chaged to RTL

alfonsoramirzz committed:
Snapshots were added to tracking

alfonsoramirzz committed:
Wrapper component was added

alfonsoramirzz committed:
- Files were changed to styled components.
- Test files were added.

alfonsoramirzz committed:
- Search file was changed to styled component
- Test file for Search was added,

alfonsoramirzz committed:
Uncomment lines

alfonsoramirzz committed:
- Home was changed to styled components.
- Home snapshot was added to tracking.
- Home test was added.
